spdxexp/satisfies_test.go

```diff
@@ -37,6 +37,28 @@ func TestValidateLicenses(t *testing.T) {
 	}
 }
 
+func TestExtractLicenses(t *testing.T) {
+	tests := []struct {
+		name              string
+		inputExpression   string
+		extractedLicenses []string
+	}{
+		{"Single license", "MIT", []string{"MIT"}},
+		{"AND'ed licenses", "MIT AND Apache-2.0", []string{"MIT", "Apache-2.0"}},
+		{"AND'ed & OR'ed licenses", "(MIT AND Apache-2.0) OR GPL-3.0", []string{"GPL-3.0", "MIT", "Apache-2.0"}},
+		{"ONLY modifiers", "LGPL-2.1-only OR MIT OR BSD-3-Clause", []string{"MIT", "BSD-3-Clause", "LGPL-2.1-only"}},
+		{"WITH modifiers", "GPL-2.0-or-later WITH Bison-exception-2.2", []string{"GPL-2.0-or-later"}},
```

Comment:
Huh, that brings up an interesting question about whether exceptions should be preserved in some fashion. It probably doesn't make a huge difference in what we need this for now, but it is relevant in general to know that a piece of code is licensed with exceptions.

I wonder if it's worth considering the license to be `GPL-2.0-or-later WITH Bison-exception-2.2` instead of just the GPL part, but I'm also wary of complicating what we're working on. Probably something to think about in a followup.

Reply:
I'm wondering if `reconstructedLicenseString()` in `node.go` has a role in this.  The current approach appears to use the simple license ignoring LicenseRed, DocumentRef, and Exceptions.  That takes a node and returns the license string with license ref, document ref, and exceptions put back in the string.

case
```
		{"WITH modifiers", "GPL-2.0-or-later WITH Bison-exception-2.2", []string{"GPL-2.0-or-later"}},
```

would become
```
		{"WITH modifiers", "GPL-2.0-or-later WITH Bison-exception-2.2", []string{"GPL-2.0-or-later WITH Bison-exception-2.2"}},
```

Reply:
I think this is a better way to go. It's more accurate to think of the license+exceptions as a different license.

Reply:
Updated it to include modifiers like above